### ISSUE TITLE
Failing test_enum_class with Python 3.13.0a5

### ISSUE THREAD

**Issue by befeleme** [sphinx-doc/sphinx#12202]
### Describe the bug

I try to build sphinx in Fedora Linux 41 with Python 3.13.0a5.
`test_enum_class` fails.
repr of the Enum class has changed. Now there's `names=<not given>` instead of None. This is an expected change, see: https://github.com/python/cpython/pull/116508#issuecomment-2015319193

```
_______________________________ test_enum_class ________________________________

app = <SphinxTestApp buildername='html'>

    @pytest.mark.sphinx('html', testroot='ext-autodoc')
    def test_enum_class(app):
        options = {"members": None}
        actual = do_autodoc(app, 'class', 'target.enums.EnumCls', options)
    
        if sys.version_info[:2] >= (3, 12):
            args = ('(value, names=None, *values, module=None, '
                    'qualname=None, type=None, start=1, boundary=None)')
        elif sys.version_info[:2] >= (3, 11):
            args = ('(value, names=None, *, module=None, qualname=None, '
                    'type=None, start=1, boundary=None)')
        else:
            args = '(value)'
    
>       assert list(actual) == [
            '',
            '.. py:class:: EnumCls' + args,
            '   :module: target.enums',
            '',
            '   this is enum class',
            '',
            '',
            '   .. py:method:: EnumCls.say_goodbye()',
            '      :module: target.enums',
            '      :classmethod:',
            '',
            '      a classmethod says good-bye to you.',
            '',
            '',
            '   .. py:method:: EnumCls.say_hello()',
            '      :module: target.enums',
            '',
            '      a method says hello to you.',
            '',
            '',
            '   .. py:attribute:: EnumCls.val1',
            '      :module: target.enums',
            '      :value: 12',
            '',
            '      doc for val1',
            '',
            '',
            '   .. py:attribute:: EnumCls.val2',
            '      :module: target.enums',
            '      :value: 23',
            '',
            '      doc for val2',
            '',
            '',
            '   .. py:attribute:: EnumCls.val3',
            '      :module: target.enums',
            '      :value: 34',
            '',
            '      doc for val3',
            '',
        ]
E       AssertionError: assert ['', '.. py:c...ass', '', ...] == ['', '.. py:c...ass', '', ...]
E         At index 1 diff: '.. py:class:: EnumCls(value, names=<not given>, *values, module=None, qualname=None, type=None, start=1, boundary=None)' != '.. py:class:: EnumCls(value, names=None, *values, module=None, qualname=None, type=None, start=1, boundary=None)'
```

### How to Reproduce

Probably by running tests with Python 3.13.0a5, but I haven't tried it on the upstream code.

### Environment Information

```text
Sphinx 7.2.6
```

**Comment by picnixz** [comment]
Closing the issue since it's fixed on HEAD (even if we did backports, I cannot make a pypi release :frowning_face:)

**Comment by jayaddison** [comment]
This seems to be occurring again, but only for a small number of our CI jobs.  For example, there's a failure in `test_enum_class` here: https://github.com/sphinx-doc/sphinx/actions/runs/8714147101/job/23903993563#step:9:2869

The change from da0733b338d1eb1f5b2c5488c36e969d44694906 includes logic related to producing `<not given>` formatting for parameters in Py3.13+ ; I'm investigating that to see whether it is relevant for the failing CI pipelines.

**Comment by jayaddison** [comment]
Ok; my current theory is that the relevant argument formatting -- I think an `enum` sentinel default value -- could have been backported to the recently-released Python 3.12.3 ; why that has only re-appeared post-7.3 release, I'm not yet certain.

**Comment by jayaddison** [comment]
A deadsnakes build from the same batch of CI jobs, passed when using [Python 3.12.2](https://github.com/sphinx-doc/sphinx/actions/runs/8714147101/job/23903993738#step:3:14) when a similar job [failed using Python 3.12.3](https://github.com/sphinx-doc/sphinx/actions/runs/8714147101/job/23903993563#step:3:14).

As far as I can tell, both CI jobs used ["3.12" as the Python version specification](https://github.com/sphinx-doc/sphinx/blob/8af2c43ca526c710e1835f3c8dc65bcbcf880885/.github/workflows/main.yml#L44-L48) - initially that makes me wonder whether `deadsnakes` could serve differing versions (e.g. 3.12.2, 3.12.3) of those at job-time.
